backend/app/services/experiment.py

```diff
@@ -164,36 +138,115 @@ async def schedule_image_building(self) -> None:
 
     async def execute_experiment_run(self, exp_run_id: PydanticObjectId) -> None:
         async with self.experiment_semaphore:
-            experiment_run, experiment = await self.init_run(exp_run_id)
-
-            error_msg = await ReanaService.run_workflow(experiment_run, experiment)
-            await ReanaService.postprocess_workflow(experiment_run, error_msg)
-
-            if not error_msg:
-                experiment_run.update_state(RunState.FINISHED)
-                await experiment_run.replace()
-            elif experiment_run.retry_count < settings.MAX_EXPERIMENT_RUN_ATTEMPTS - 1:
-                experiment_run.update_state(RunState.CRASHED)
-                new_exp_run = experiment_run.retry_failed_run()
-
-                await new_exp_run.create()
-                await experiment_run.replace()
-                await self.add_run_to_execute(new_exp_run.id)
-            else:
-                experiment_run.update_state(RunState.CRASHED)
-                await experiment_run.replace()
+            try:
+                experiment_run, experiment = await self.init_run(exp_run_id)
+                if experiment_run is None:
```

Comment:
What does this case mean? Don't we need to do some action here?

Reply:
necessary actions are performed within the `init_run` function. And this use case describes a situation in which we firstly couldn't find a docker image for corresponding template and we further didnt succeed in rebuilding it either

Reply:
Should we at least log something?

Reply:
sure, it wont hurt